### PR TITLE
    feat(table): added support for single row radio select

### LINKED ISSUE #4882: Add radio button select in Table
Introduce a new Table variant that supports radio button select rather than checkbox select to select only one row at a time. Core issue https://github.com/patternfly/patternfly/issues/3418

### PR DESCRIPTION
**Pull request by KKoukiou**
Closes #4882

### CODE REVIEW

File: packages/react-table/src/components/Table/examples/Table.md [hunk 1]
```diff
@@ -487,6 +487,84 @@ class SelectableTable extends React.Component {
 }
 ```
 
+### Selectable Radio Input
```
Comment (redallen) [inline packages/react-table/src/components/Table/examples/Table.md:490]: Use sentence case: `Selectable radio input`

**Review by redallen — CHANGES_REQUESTED** [review]
Probably don't want a radio button in the table header?
image

**Comment by mcoker** [comment]
Aside from @redallen's comments, this LGTM. One thing I noticed is that the react table uses `pf-c-check__input` and `pf-c-radio__input` for the check and radio inputs, but we don't use those in core. They're just regular `<input type="[radio/check]">` elements. The check and radio components are designed to work as an input/label pair, and the `pf-c-[check/radio]__input` elements are expected to be children of `pf-c-[check/radio]`.

I don't think that we want to change the checkbox inputs since that could break users if they're styling them using the `pf-c-check__input` class or something like that, but I'm wondering if we should use regular radio inputs, or use the `pf-c-radio__input` element to be consistent with the table checkboxes.

**Comment by KKoukiou** [comment]
> Aside from @redallen's comments, this LGTM. One thing I noticed is that the react table uses `pf-c-check__input` and `pf-c-radio__input` for the check and radio inputs, but we don't use those in core. They're just regular `<input type="[radio/check]">` elements. The check and radio components are designed to work as an input/label pair, and the `pf-c-[check/radio]__input` elements are expected to be children of `pf-c-[check/radio]`.
> 
> I don't think that we want to change the checkbox inputs since that could break users if they're styling them using the `pf-c-check__input` class or something like that, but I'm wondering if we should use regular radio inputs, or use the `pf-c-radio__input` element to be consistent with the table checkboxes.

@mcarrano amazing how you noticed this! Indeed I introduced these classes in a separate commit in this PR, and the reason I did it, it because I noticed that checboxes/radio inputs without these two classes (`pf-c-radio__input, pf-c-check__input`) do not show the black circle when being hovered over in a disable mode.

This is a bug and adding these classes fixes the issue.

So if I was to choose between fixing an existing bug by using the classes, and protecting users who are using custom CSS on these classes and might see issues, I would go with the first.
In any case, I leave it up to you to decide :)

File: packages/react-table/src/components/Table/Table.tsx [hunk 2]
```diff
@@ -237,10 +243,12 @@ export interface IRow extends RowType {
   isExpanded?: boolean;
   isFirstVisible?: boolean;
   isLastVisible?: boolean;
-  /* Whether the row checkbox is selected */
+  /** Whether the row checkbox/radio button is selected */
   selected?: boolean;
-  /* Whether the row checkbox is disabled */
+  /** deprecated - Use disableSelection instead - Whether the row checkbox is disabled */
   disableCheckbox?: boolean;
+  /** Where the row checkbox/radio button is disabled */
```
Comment (jschuler) [inline packages/react-table/src/components/Table/Table.tsx:250]:
```suggestion
  /** Whether the row checkbox/radio button is disabled */
```

**Review by redallen — APPROVED** [review]
Code LGTM!

**Comment by mcoker** [comment]
@KKoukiou ah! Thanks for the explanation and fixing the bug. It's ultimately a misuse of the radio/check component, and will also change the positioning of the existing check input because the check component's input has a margin - https://github.com/patternfly/patternfly/blob/1b3c31608b8d7f43f3509cdeef8b7e62358b8a46/src/patternfly/components/Check/check.scss#L26-L28

I'm in favor of leaving it with the disabled input bug for the time being. We have an issue to allow the check/radio inputs to be used on their own in the check/radio components here - https://github.com/patternfly/patternfly/issues/3311

I think once we have support for  https://github.com/patternfly/patternfly/issues/3311 we can look at whether we should introduce the check/radio component input here, or just style the regular check/radio inputs in the table to support a disabled state until we're able to update the table to use the check/radio component inputs. Sound ok?

**Review by jschuler — COMMENTED** [review]
Code LGTM, just need to clarify on the radio/checkbox styling with @mcoker

**Comment by KKoukiou** [comment]
> @KKoukiou ah! Thanks for the explanation and fixing the bug. It's ultimately a misuse of the radio/check component, and will also change the positioning of the existing check input because the check component's input has a margin - https://github.com/patternfly/patternfly/blob/1b3c31608b8d7f43f3509cdeef8b7e62358b8a46/src/patternfly/components/Check/check.scss#L26-L28
> 
> I'm in favor of leaving it with the disabled input bug for the time being. We have an issue to allow the check/radio inputs to be used on their own in the check/radio components here - [patternfly/patternfly#3311](https://github.com/patternfly/patternfly/issues/3311)
> 
> I think once we have support for [patternfly/patternfly#3311](https://github.com/patternfly/patternfly/issues/3311) we can look at whether we should introduce the check/radio component input here, or just style the regular check/radio inputs in the table to support a disabled state until we're able to update the table to use the check/radio component inputs. Sound ok?

Perfect! Reverting the relevant commit.

**Review by mcoker — APPROVED** [review]
lgtm! Thanks @KKoukiou!!

**Review by mcoker — APPROVED** [review]

**Review by redallen — APPROVED** [review]

**Pull request merged**